### PR TITLE
Add reply

### LINKED ISSUE #9: Post a reply for a thread

### PR DESCRIPTION
**Pull request by Eedwarddi**
Fixes #9 

Created a new endpoint `/threads/{threadId}/replies` to allow posting replies to threads

URL example for POST method:

`http://localhost:8080/web-forum/api/threads/67/replies`

Body example for the call:

```
{	
	"comment" : "This is a comment of the reply",
}
```

Example of what the server returns:

```
{
    "id": 2,
    "comment": "This is a comment of the reply",
    "thread": {
        "id": 67,
        "title": "This a title for a thread",
        "content": "Content of the thread",
        "category": {
            "id": 3,
            "title": "Video Games"
        }
    }
}
```

### CODE REVIEW

**Comment by henrikra** [comment]
If you check how you can add tracks to playlist with Spotify API https://developer.spotify.com/documentation/web-api/reference/playlists/ then your endpoint should be `/threads/{thread_id}/replies`. What do you think?

File: src/main/java/forum/dao/ReplyDAOImpl.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	
+	@Autowired
+	private SessionFactory sessionFactory;
+
+	@Override
+	public int saveReply(Reply reply) {
+		
+		Session currentSession = sessionFactory.getCurrentSession();
+		
+		int result = (int) currentSession.save(reply);
```
Comment (henrikra) [inline src/main/java/forum/dao/ReplyDAOImpl.java]: Can this variable name be more specific? Like `replyId` etc :D

File: src/main/java/forum/service/ReplyServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	@Autowired
+	private ThreadService threadService;
+
+	@Override
+	@Transactional
+	public Reply saveReply(Reply reply) {
+		
+		Thread tempThread = threadService.getThread(reply.getThread().getId());
+		
+		tempThread.add(reply);
```
Comment (henrikra) [inline src/main/java/forum/service/ReplyServiceImpl.java]: How the app behaves if you don't execute this line?

**Review by henrikra — COMMENTED** [review]
Couple questions listed 👍

**Comment by Eedwarddi** [comment]
> If you check how you can add tracks to playlist with Spotify API https://developer.spotify.com/documentation/web-api/reference/playlists/ then your endpoint should be `/threads/{thread_id}/replies`. What do you think?

I can also do it that way, maybe it is better? Should I also do it that way for posting threads for a category also?

File: src/main/java/forum/dao/ReplyDAOImpl.java (hunk 1 above)
Comment (Eedwarddi) [reply src/main/java/forum/dao/ReplyDAOImpl.java]: Yes, you are right. I renamed the variable.

**Comment by henrikra** [comment]
Yeah sounds good 👍

File: src/main/java/forum/service/ReplyServiceImpl.java (hunk 2 above)
Comment (Eedwarddi) [reply src/main/java/forum/service/ReplyServiceImpl.java]: This is what the server returns for the POST call when this line is not executed:

```
{
    "id": 3,
    "comment": "A comment for a thread",
    "thread": {
        "id": 67,
        "title": null,
        "content": null,
        "category": null
    }
}
```

**Review by henrikra — APPROVED** [review]
Looks good. Just add error handling to the endpoints 👍

**Pull request merged**